### PR TITLE
[MAC] - ScreenReader support on submit of cards with error message

### PR DESCRIPTION
**Pull request by snbehar**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

## Sample Card
ezgif-5-85b0f8cb92

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/macos/AdaptiveCards/AdaptiveCards/Base/BaseProtocol.swift [hunk 1]
```diff
@@ -34,11 +34,13 @@ protocol InputHandlingViewProtocol: NSView {
     var isRequired: Bool { get }
     var errorDelegate: InputHandlingViewErrorDelegate? { get set }
     func showError()
+    func setFocus()
 }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Base/BaseProtocol.swift:38]:
```suggestion
    func setAccessibilityFocus()
}
```

File: source/macos/AdaptiveCards/AdaptiveCards/Base/BaseProtocol.swift [hunk 2]
```diff
@@ -34,11 +34,13 @@ protocol InputHandlingViewProtocol: NSView {
...
 
 protocol InputHandlingViewErrorDelegate: AnyObject {
     func inputHandlingViewShouldShowError(_ view: InputHandlingViewProtocol)
     func inputHandlingViewShouldHideError(_ view: InputHandlingViewProtocol, currentFocussedView: NSView?)
+    func inputHandlingViewShouldAnnounceErrorMessage()
     var isErrorVisible: Bool { get }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Base/BaseProtocol.swift:44]:
```suggestion
    func inputHandlingViewShouldAnnounceErrorMessage(_ view: InputHandlingViewProtocol)
    var isErrorVisible: Bool { get }
```

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift [hunk 3]
```diff
@@ -309,6 +309,10 @@ extension ACRContentStackView: InputHandlingViewErrorDelegate {
         hideErrorMessage(with: currentFocussedView)
     }
     
+    func inputHandlingViewShouldAnnounceErrorMessage() {
+        NSAccessibility.announce("Error! " + (errorMessageField?.stringValue ?? "") + ". " + (inputLabelField?.stringValue ?? ""))
+    }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift:314]: Is this tested?
Just curious how is "Error '!'" is read. Does it explicitly say "exclamation" or just read Error?

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRView.swift [hunk 4]
```diff
@@ -80,6 +82,8 @@ class ACRView: ACRColumnView {
             // If it is a textfield, the entered text gets selected when focus is set, so taking the cursor to the last position it was present
             guard let focusedTextField = focusedElementOnHideError as? ACRTextField else { return }
             focusedTextField.resetCursorPositionIfNeeded()
+        } else if firstFieldWithError != nil {
+            firstFieldWithError?.setFocus()
         }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRView.swift:87]: Why is this included inside keyboard reset flow?
This is something related to ScreenReaders only right?

**Review by blue-stone99 — COMMENTED** [review]
Add tests for these..

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift (hunk 3 above)
Comment (snbehar) [reply source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift:314]: It just says Error

**Review by mukuagar — APPROVED** [review]
Overall code look fine to me
But you'll have to test on webex to see the changes work as expected reliably
